src/pages/storage/forms/StorageBucketKeyForm.tsx

```diff
@@ -0,0 +1,99 @@
+import type { FC, ReactNode } from "react";
+import { Form, Input, Select } from "@canonical/react-components";
+import type { FormikProps } from "formik/dist/types";
+import AutoExpandingTextArea from "components/AutoExpandingTextArea";
+import { useStorageBucketEntitlements } from "util/entitlements/storage-buckets";
+import type { LxdStorageBucket } from "types/storage";
+
+export interface StorageBucketKeyFormValues {
+  name: string;
+  role?: string;
+  description?: string;
+  "access-key"?: string;
+  "secret-key"?: string;
+}
+
+interface Props {
+  formik: FormikProps<StorageBucketKeyFormValues>;
+  bucket?: LxdStorageBucket;
+}
+
+const StorageBucketKeyForm: FC<Props> = ({ formik, bucket }) => {
+  const { canEditBucket } = useStorageBucketEntitlements();
+  const getFormProps = (
+    id: "name" | "description" | "access-key" | "secret-key",
+  ) => {
+    return {
+      id: id,
+      name: id,
+      onBlur: formik.handleBlur,
+      onChange: formik.handleChange,
+      value: formik.values[id] ?? "",
+      error: formik.touched[id] ? (formik.errors[id] as ReactNode) : null,
+      placeholder: `Enter ${id.replaceAll("-", " ")}`,
+    };
+  };
+
+  const isEditing = !!bucket;
+
+  const bucketEditRestriction =
+    !isEditing || canEditBucket(bucket)
+      ? ""
+      : "You do not have permission to modify this bucket";
```

Comment:
```suggestion
      : "You do not have permission to edit this bucket";
```
I think we should be consistent with "edit".